lib/core/src/can_sendable.rs

```diff
@@ -0,0 +1,292 @@
+use defmt::error;
+
+pub const CAN_MSG_TYPE_BOOL: u8 = 0;
+pub const CAN_MSG_TYPE_F32: u8 = 1;
+pub const CAN_MSG_TYPE_TWO_U16: u8 = 2;
+pub const CAN_MSG_TYPE_POSDELTA: u8 = 3;
+
+/// Checks if a CAN message is a valid message
+pub fn is_valid_can_msg(msg: &[u8; 8]) -> bool {
+    // we assume that board id is valid as ATM we dont know how many boards sending messages there are
+    (msg[0] & 0x0F) < 4
+}
+
+/// Extracts the message type from a CAN message
+pub fn can_msg_type_from_u8(msg: &[u8; 8]) -> CanMsgType {
+    match msg[0] & 0x0F {
+        CAN_MSG_TYPE_BOOL => CanMsgType::Bool,
+        CAN_MSG_TYPE_F32 => CanMsgType::F32,
+        CAN_MSG_TYPE_TWO_U16 => CanMsgType::U32,
+        CAN_MSG_TYPE_POSDELTA => CanMsgType::PosDelta,
+        _ => {
+            error!("Unknown CAN message type: {}", msg[0] & 0x0F);
+            panic!();
+        }
+    }
+}
+
+/// Extracts the board id from a CAN message
+pub fn get_board_id(msg: &[u8; 8]) -> u8 {
+    msg[0] >> 4
+}
+
+/// Builds a CAN header byte from a board id and message type
+pub fn build_can_header(board_id: u8, msg_type: u8) -> u8 {
+    (board_id << 4) | msg_type
+}
+
+pub enum CanMsgType {
+    Bool,
+    F32,
+    U32,
+    PosDelta,
+}
+
+pub trait CanSendable {
+    fn encode_to_can(&self, board_id: u8) -> [u8; 8];
+    fn can_decode(can_msg: &[u8; 8]) -> Self;
+}
+
+impl CanSendable for bool {
+    fn encode_to_can(&self, board_id: u8) -> [u8; 8] {
+        let mut data: [u8; 8] = [0; 8];
+        data[0] = build_can_header(board_id, CAN_MSG_TYPE_BOOL);
```

Comment:
This should be an enum becuase there's only the specific set values that have meaning.